odoo/addons/base/models/res_currency.py

```diff
@@ -247,6 +247,29 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
+    @api.onchange('rate')
+    def _onchange_rate_warning(self):
+        if not self.currency_id.id:
+            return
+
+        latest_rate = self.search([
+            ('currency_id', '=', self.currency_id.id),
+            ('company_id', '=', self.company_id.id or self.env.company.id),
+            ('name', '<=', self.name or fields.Date.today()),
+        ], order="name desc", limit=1)
+        if latest_rate:
+            diff = (latest_rate.rate - self.rate) / latest_rate.rate
+            if abs(diff) > 0.2:
+                return {
+                    'warning': {
+                        'title': _("Warning for %s", self.currency_id.name),
+                        'message': _(
+                            "The rate you've set is quite different than the previous rate.\n"
```

Comment:
*different from* + avoid addressing user as "you".

```suggestion
                            "The new rate is quite far from the previous rate.\n"
```